lib/thrift/include/thrift/transport/TSSLSocket.h

```diff
@@ -0,0 +1,428 @@
+/*
+ * Copyright 2006 Facebook
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _THRIFT_TRANSPORT_TSSLSOCKET_H_
+#define _THRIFT_TRANSPORT_TSSLSOCKET_H_ 1
+
+// Put this first to avoid WIN32 build failure
+#include <thrift/transport/TSocket.h>
+
+// #include <openssl/ssl.h>
+#include <string>
+#include <thrift/concurrency/Mutex.h>
+
+#include <zephyr/net/tls_credentials.h>
+#include <zephyr/posix/sys/socket.h>
+
+namespace apache {
+namespace thrift {
+namespace transport {
+
+class AccessManager;
+#if 0
+class SSLContext;
+#endif
+
+enum SSLProtocol {
+  SSLTLS = 0,  // Supports SSLv2 and SSLv3 handshake but only negotiates at TLSv1_0 or later.
+               // SSLv2   = 1,  // HORRIBLY INSECURE!
+  SSLv3 = 2,   // Supports SSLv3 only - also horribly insecure!
+  TLSv1_0 = 3, // Supports TLSv1_0 or later.
+  TLSv1_1 = 4, // Supports TLSv1_1 or later.
+  TLSv1_2 = 5, // Supports TLSv1_2 or later.
+  LATEST = TLSv1_2
+};
+
+enum ThriftTLScertificateType {
```

Comment:
`ThriftTLScertificateType` can probably just be declared inside `TSSLServerSocket.cpp`